src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,96 @@
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Field[] declaredField = clazz.getDeclaredFields();
+
+        for (Field field: declaredField) {
+
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                clazzImplementationInstance = createNewInstance(clazz);
+                field.setAccessible(true);
+                try {
+                    field.set(clazzImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't initialize field value."
+                            + "Class: " + clazz.getName()
+                            + ". Field: " + field.getName() + e);
+                }
+            }
+        }
+
+        if (clazzImplementationInstance == null) {
+            clazzImplementationInstance = createNewInstance(clazz);
+        }
+        return clazzImplementationInstance;
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+
+            if (!constructor.canAccess(null)) {
+                constructor.canAccess(true);
+            }
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (NoSuchMethodException e) {
+            throw new RuntimeException("Class " + clazz.getName()
+                    + " must have no-argument constructor", e);
+        } catch (InvocationTargetException
+                 | InstantiationException
+                 | IllegalAccessException e) {
+            throw new RuntimeException("Can't create a new instance of " + clazz.getName() + e);
+        }
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
+        Map<Class<?>, Class<?>> interfaceImplementations = new HashMap<>();
+        interfaceImplementations.put(FileReaderService.class, FileReaderServiceImpl.class);
+        interfaceImplementations.put(ProductParser.class, ProductParserImpl.class);
+        interfaceImplementations.put(ProductService.class, ProductServiceImpl.class);
+
+        Class<?> clazzImplementation = interfaceClazz;
+
+        if (interfaceClazz.isInterface()) {
+            clazzImplementation = interfaceImplementations.get(interfaceClazz);
+
+            if (clazzImplementation == null) {
+                throw new RuntimeException("There is no implementation for: " + interfaceClazz);
```

Comment:
The `findImplementation` method throws a `RuntimeException` if no implementation is found, which is correct. However, ensure that this exception is handled appropriately where `getInstance` is called to avoid unexpected crashes.